src/vs/workbench/api/browser/mainThreadWebviews.ts

```diff
@@ -89,7 +89,7 @@ export class MainThreadWebviews extends Disposable implements extHostProtocol.Ma
 	private onDidClickLink(handle: extHostProtocol.WebviewHandle, link: string): void {
 		const webview = this.getWebview(handle);
 		if (this.isSupportedLink(webview, URI.parse(link))) {
-			this._openerService.open(link, { fromUserGesture: true, allowContributedOpeners: true, allowCommands: true });
+			this._openerService.open(link, { fromUserGesture: true, allowContributedOpeners: true, allowCommands: true, fromWorkspace: true });
```

Comment:
For webviews, I propose:

- The project release notes should always set `fromWorkspace: true` since we control the release notes
- The back layer notebook should also set `fromWorkspace: true` since the file is coming from the workspace 
- We also treat all webviews created by extensions as `fromWorkspace: true`. This is not accurate but it's also not possible to know if the webview is just showing content from the workspace or content from somewhere else
- We don't set `fromWorkspace` for the extension pages

Reply:
I think for project release notes we should not set `fromWorkspace` to `true`. Yes, we control the release notes but they are not from the workspace. They should be validated by trusted domains which has some preconfiguration to trust domains we own.

Reply:
I've made the change for notebook webviews and I think the extension webviews were handled by the previous commits. I've tested:
- Release notes
- Extension Pages
- Notebook input editors / outputs and markdown outputs
- Pull Request description page links
- Terminal links
- Editor links